include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -7782,12 +7783,7 @@ namespace VectorTools
     else
       {
         // This function is not implemented for distributed vectors.
-        Assert(typeid(v) == typeid(Vector<double>) ||
-               typeid(v) == typeid(Vector<float>) ||
-               typeid(v) == typeid(BlockVector<double>) ||
-               typeid(v) == typeid(BlockVector<float>) ||
-               typeid(v) == typeid(LinearAlgebra::Vector<double>) ||
-               typeid(v) == typeid(LinearAlgebra::Vector<float>),
+        Assert(dealii::is_serial_vector< VectorType >::value,
```

Comment:
make it more obvious by saying `...::value == true` -- by the time I've read all the way down to the `value`, I've forgotten already that the class name contains a predicate (`is_..`).